### DISCUSSION TITLE
Convert string from Text to str keeping ansi escape codes intact

### DISCUSSION THREAD

**Discussion by iwconfig** [Textualize/rich#2648] [Q&A] [answered]
Hello!

I've been wandering around the roads of the internet for hours. Is it possible to use `rich.text.Text` to return a `str` of the formatted text? Like an inverse of `Text.from_ansi` that would look something like this:

```python3
>>> s = Text.to_ansi('[b]This is bold text.')
>>> str(s)
'\x1b[1mThis is bold text.\x1b[0m'
```

**Comment by iwconfig** [comment]
OK I found out one can do

```python3
>>> S = Style(bold=True)
>>> s = S.render('This is bold text.')
>>> str(s)
'\x1b[1mThis is bold text.\x1b[0m'
```

I suppose that could work *but I'd prefer using markup*

**Comment by willmcgugan** [comment, marked as answer]
See the docs for how to [capture](https://rich.readthedocs.io/en/latest/console.html#capturing-output) output.

**Comment by iwconfig** [comment]
Oh! How could I miss that. I thought of `Console.capture` as rich version of some sort of `builtins.input` kind of thing. I skimmed through to fast I guess. Oh well!

```python3
from rich.console import Console
console = Console()

def text_to_ansi(s):
    with console.capture() as capture:
        console.print(s)
    return capture.get()

s = text_to_ansi('[b]This is bold text.')
str(s)
# '\x1b[1mThis is bold text.\x1b[0m\n'
```

Easy! Thank you so much Will! I have to say, I'm so impressed with *Rich* and ***Textual***. <ins>***Such cool tools.***</ins> 🤩
